src/shared/telemetry/telemetryEvent.ts

```diff
@@ -7,7 +7,7 @@
 
 import { MetadataEntry, MetricDatum, Unit } from './clienttelemetry'
 
-const NAME_ILLEGAL_CHARS_REGEX = new RegExp('[^\\w+-.:]', 'g')
+const NAME_ILLEGAL_CHARS_REGEX = new RegExp('[^\\w+-.:_]', 'g')
```

Comment:
On line 44 you have to apply `NAME_ILLEGAL_CHARS_REGEX` to the two portions before joining them with a _ because that will get stripped out. I'm surprised a test did not catch this, please add a test before making the fix.

Reply:
I'll give it a look. I'm not sure if we have any tests on the sent data.

Reply:
Looks like `toMetricData` can have tests written for this.

Reply:
So a few things to take into consideration:
* The Java definition for this regex comes from https://github.com/aws/aws-toolkit-jetbrains/blob/master/telemetry-client/telemetryC2J/service-2.json#L123 
* `\w` includes alphanumeric characters AND underscores and cannot be rigged to remove the underscores.

In the interest of not changing our service definition, I'll add a separate regex check to remove underscores from the namespace and name themselves, and then pass that through this existing filter. The equivalent is `new RegExp('[^a-zA-Z0-9+-.:]', 'g')`, which will deviate from the service definition.